pkg/asset/installconfig/aws/validation_test.go

```diff
@@ -472,7 +566,9 @@ func TestValidate(t *testing.T) {
 			if test.expectErr == "" {
 				assert.NoError(t, err)
 			} else {
-				assert.Regexp(t, test.expectErr, err.Error())
+				if assert.Error(t, err) {
+					assert.Regexp(t, test.expectErr, err.Error())
+				}
```

Comment:
Can we flatten this... `} else if {`

Reply:
I would prefer not to. The current layout makes a distinction between the codepath where we *are not* expecting an error and the codepath where we *are* expecting an error. If we flatten to an `else if`, that makes the two codepaths are instead (1) we are not expecting an error and (2) we are expecting an error and we actually got an error.
```
if test.expectErr == "" {
   // assert that we did not get an error
} else {
   // assert that we got an error and that the error matches what we expected
}
```
versus
```
if test.expectErr == "" {
   // assert that we did not get an error
} else if assert.Error(t, err) {
   // assert that the error that we got matches what we expected
}
```